sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/CountAggregator.java

```diff
@@ -8,54 +8,62 @@
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
 import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.Exemplar;
 import io.opentelemetry.sdk.metrics.data.LongSumData;
 import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.exemplar.ExemplarReservoir;
 import io.opentelemetry.sdk.metrics.internal.descriptor.MetricDescriptor;
 import io.opentelemetry.sdk.resources.Resource;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.atomic.LongAdder;
+import java.util.function.Supplier;
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
-final class CountAggregator extends AbstractAggregator<Long> {
+final class CountAggregator extends AbstractAggregator<LongAccumulation> {
   private final AggregationTemporality temporality;
+  // Workaround
+  private final Supplier<ExemplarReservoir> reservoirSupplier;
 
   CountAggregator(
       Resource resource,
       InstrumentationLibraryInfo instrumentationLibraryInfo,
       MetricDescriptor descriptor,
-      AggregationTemporality temporality) {
+      AggregationTemporality temporality,
+      Supplier<ExemplarReservoir> reservoirSupplier) {
     super(
         resource,
         instrumentationLibraryInfo,
         descriptor,
         temporality == AggregationTemporality.CUMULATIVE);
     this.temporality = temporality;
+    this.reservoirSupplier = reservoirSupplier;
   }
 
   @Override
-  public AggregatorHandle<Long> createHandle() {
-    return new Handle();
+  public AggregatorHandle<LongAccumulation> createHandle() {
+    return new Handle(reservoirSupplier.get());
   }
 
   @Override
-  public Long accumulateDouble(double value) {
-    return 1L;
+  public LongAccumulation accumulateDouble(double value) {
+    return LongAccumulation.create(1L);
   }
 
   @Override
-  public Long accumulateLong(long value) {
-    return 1L;
+  public LongAccumulation accumulateLong(long value) {
+    return LongAccumulation.create(1L);
   }
 
   @Override
-  public Long merge(Long a1, Long a2) {
-    return a1 + a2;
+  public LongAccumulation merge(LongAccumulation a1, LongAccumulation a2) {
+    return LongAccumulation.create(a1.getValue() + a2.getValue(), a2.getExemplars());
```

Comment:
why not the set of both a1 and a2's exemplars? I'm not sure I would have known this was the right way to implement `merge` with exemplars in play.

Reply:
Two main points:

1. I'm generally not happy with this particular implementation here.  Without doing a refactor of the Aggregator interface, there's a lot of subtle details in methods that are either not documented (or buggy, e.g. how merge is used between Sync + Async aggregation w/ Sum).   a1 = previous, a2 = current when exemplars exist.   For Async instruments (where you'd want to merge), they'd both be empty.
2. According to the spec, it's best-effort to preserve exemplars (on purpose) so we can optimise code paths.  Additionally, for cumulative instruments, it's expected you show recent exemplars.  There is no expectation of perserverance.

If you think it's worth it (in this CL) to do some light refactoring + documentation of the merge method, I'd be more than happy to kick that off.  I'd been avoiding anything major until all the pieces are in place.   However, it's really getting unwieldy.

Reply:
This is fine for now. As long as we're doing it consistently, that's a good start.